bolt-contracts/src/contracts/BoltManager.sol

```diff
@@ -0,0 +1,287 @@
+        return validators.getValidatorByPubkeyHash(pubkeyHash).authorizedOperator == operator;
+    }
+
+    /// @notice Returns the addresses of the middleware contracts of restaking protocols supported by Bolt.
+    function getSupportedRestakingProtocols() public view returns (address[] memory middlewares) {
+        return restakingProtocols.values();
+    }
+
+    /// @notice Returns whether an operator is registered with Bolt.
+    function isOperator(
+        address operator
+    ) public view returns (bool) {
+        return operators.contains(operator);
+    }
+
+    /// @notice Get the status of multiple proposers, given their pubkey hashes.
+    /// @param pubkeyHashes The pubkey hashes of the proposers to get the status for.
+    /// @return statuses The statuses of the proposers, including their operator and active stake.
+    function getProposerStatuses(
+        bytes32[] calldata pubkeyHashes
+    ) public view returns (IBoltValidators.ProposerStatus[] memory statuses) {
+        statuses = new IBoltValidators.ProposerStatus[](pubkeyHashes.length);
+        for (uint256 i = 0; i < pubkeyHashes.length; ++i) {
+            statuses[i] = getProposerStatus(pubkeyHashes[i]);
+        }
+    }
+
+    /// @notice Get the status of a proposer, given their pubkey hash.
+    /// @param pubkeyHash The pubkey hash of the proposer to get the status for.
+    /// @return status The status of the proposer, including their operator and active stake.
+    function getProposerStatus(
+        bytes32 pubkeyHash
+    ) public view returns (IBoltValidators.ProposerStatus memory status) {
+        if (pubkeyHash == bytes32(0)) {
+            revert InvalidQuery();
+        }
+
+        uint48 epochStartTs = getEpochStartTs(getEpochAtTs(Time.timestamp()));
+        // NOTE: this will revert when the proposer does not exist.
+        IBoltValidators.Validator memory validator = validators.getValidatorByPubkeyHash(pubkeyHash);
+
+        Operator memory operator = operators.get(validator.authorizedOperator);
+
+        status.pubkeyHash = pubkeyHash;
+        status.operator = validator.authorizedOperator;
+        status.operatorRPC = operator.rpc;
+
+        (uint48 enabledTime, uint48 disabledTime) = operators.getTimes(validator.authorizedOperator);
+        if (!_wasEnabledAt(enabledTime, disabledTime, epochStartTs)) {
+            return status;
+        }
+
+        (status.collaterals, status.amounts) =
+            IBoltMiddleware(operator.middleware).getOperatorCollaterals(validator.authorizedOperator);
+
+        // NOTE: check if the sum of the collaterals covers the minimum operator stake required.
+
+        uint256 totalOperatorStake = 0;
+        for (uint256 i = 0; i < status.amounts.length; ++i) {
+            totalOperatorStake += status.amounts[i];
+        }
+
+        if (totalOperatorStake < parameters.MINIMUM_OPERATOR_STAKE()) {
+            status.active = false;
+        } else {
+            status.active = true;
+        }
```

Comment:
Just to confirm, here we consider the total amount as sum of individual staked assets amounts, which means they should all be reasonably close one another in value (and have the same amount of decimals), right?